app/controllers/sns_controller.rb

```diff
@@ -0,0 +1,53 @@
+# These endpoints allow the Amazon SES service to notify us when there are bounced
+# emails or complaints (ex we are marked as spam). The expected JSON is particular to
+# Amazon's SNS notification service, and thus will not work for other services.
+class SnsController < ApplicationController
+  protect_from_forgery with: :null_session
+  before_filter :verify_amazon_authorize_key
+
+  def bounce
+    begin
+      success = true
+      message = JSON.parse(request.body.read)['Message']
+      bounce_arr = JSON.parse(message)['bounce']['bouncedRecipients']
+      bounce_arr.each do |recipient|
+        Bounce.create(email: recipient['emailAddress'].downcase)
+      end
+    rescue
```

Comment:
I think an exception is appropriate if the input is unexpected. Lately I've been using Sentry event contexts and finding them really useful. We could do something like this

```ruby
  before :set_context

  def set_context
    Raven.extra_context(message: request.body.read)
    request.body.rewind
  end
```

Then if there's a parse error we'll get notified and also have the post body available to inspect.